src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -1154,6 +1154,26 @@ namespace cryptonote
     return buf;
   }
   //---------------------------------------------------------------
+  bool is_valid_address(const std::string address, bool allow_subaddress, bool allow_integrated)
+  {
+    cryptonote::address_parse_info addr_info;
+    bool valid = false;
+    std::initializer_list<cryptonote::network_type> all_nettypes = {cryptonote::network_type::MAINNET, cryptonote::network_type::TESTNET, cryptonote::network_type::DEVNET, cryptonote::network_type::FAKECHAIN};
+    for (auto nettype : all_nettypes) {
```

Comment:
Where does this get used?  If it's scanning all the network types, does that mean there are places where I could (for example on some platform doing address validation) pass in a testnet address when it is expecting mainnet addresses?

Reply:
Right now its only used in 2 places
1) to guess if a value passed to `ons_buy_mapping` is a wallet type
2) enforcing that the parameter passed as owner to `ons_print_owners_to_names` is valid

Both these would fail later on if there was an address from the wrong network type used. But its also pretty easy to add a network type param here

Reply:
> But its also pretty easy to add a network type param here

I think that's a good idea, less possibility for gotchas in the future.  (I also can't really imagine a place where you'd want to allow any of the three -- the three networks are effectively entirely different chains).